be/src/runtime/table_function_table_sink.cpp

```diff
@@ -56,4 +56,75 @@ Status TableFunctionTableSink::close(RuntimeState* state, Status exec_status) {
     return Status::OK();
 }
 
+Status TableFunctionTableSink::decompose_to_pipeline(pipeline::OpFactories prev_operators, const TDataSink& thrift_sink,
+                                                     pipeline::PipelineBuilderContext* context) const {
+    auto runtime_state = context->runtime_state();
+    auto fragment_ctx = context->fragment_context();
+    auto output_exprs = this->get_output_expr();
+
+    DCHECK(thrift_sink.table_function_table_sink.__isset.target_table);
+    DCHECK(thrift_sink.table_function_table_sink.__isset.cloud_configuration);
+
+    const auto& target_table = thrift_sink.table_function_table_sink.target_table;
+    DCHECK(target_table.__isset.path);
+    DCHECK(target_table.__isset.file_format);
+    DCHECK(target_table.__isset.columns);
+    DCHECK(target_table.__isset.write_single_file);
+    DCHECK(target_table.columns.size() == output_exprs.size());
+
+    std::vector<std::string> column_names;
+    for (const auto& column : target_table.columns) {
+        column_names.push_back(column.column_name);
+    }
+
+    // prepare sink context
+    auto sink_ctx = std::make_shared<connector::FileChunkSinkContext>();
+    sink_ctx->path = target_table.path;
+    sink_ctx->cloud_conf = thrift_sink.table_function_table_sink.cloud_configuration;
+    sink_ctx->column_names = column_names;
```

Comment:
```suggestion
    sink_ctx->column_names = std::move(column_names);
```
to avoid unnecessary string copy